### DISCUSSION TITLE
pcf start and WebAPI support

### DISCUSSION THREAD

**Discussion by dmcweeney** [microsoft/powerplatform-vscode#367] [Q&A]
Hi,

Just looking for an update on what is planned (if anything) for WebAPI support in a pcf when getting launched by pcf start?

Thanks, Donal

**Comment by BetimBeja** [comment]
While not exactly the answer you are looking for. You might want to try StorybookJs with my package `@shko.online/componentframework-mock`
We have mocked all the ComponentFramework API and you can easily test your PCF components. If something is not implemented yet, everything is a mock and can be faked pretty easily.

**Comment by dmcweeney** [comment]
@BetimBeja thanks for the pointer. I'll have to investigate in a few weeks when I get a bit of time.

**Comment by dmcweeney** [comment]
@petrochuk I'm disappointed to see this marked as closed when it is still unanswered.

**Comment by petrochuk** [comment]
@dmcweeney It was labeled as 'needs more info' because it is not clear. Are you talking about this [WebAPI](https://www.w3schools.com/js/js_api_intro.asp)? What do you mean by support?

**Reply by BetimBeja** [reply]
@petrochuk @dmcweeney is talking about this https://learn.microsoft.com/power-apps/developer/component-framework/reference/webapi?WT.mc_id=DX-MVP-5004767 in the test harness you get a not implemented alert message if you try to use any of these functions.

**Comment by dmcweeney** [comment]
Hi @petrochuk,

Specifically here - [Common limitations when using the test harness](https://learn.microsoft.com/en-us/power-apps/developer/component-framework/debugging-custom-controls#common-limitations-when-using-the-test-harness) - point 2:

_Using features listed in the feature-usage section of the ControlManifest.Input.xml. For example, **calling the context.WebApi.*** methods will throw an exception from within the test harness._

Also apologies but I never spotted the 'needs more info' label.

Thanks,

Donal

cc: @BetimBeja
